File: Easy/217 - ContainsDuplicate/solution.ts
// Question 217. Contains Duplicate (https://leetcode-cn.com/problems/contains-duplicate/)

/**
 * 执行用时：96 ms, 在所有 TypeScript 提交中击败了 68.22% 的用户
 * 内存消耗：44.4 MB, 在所有 TypeScript 提交中击败了 39.51% 的用户
 *
 * @param {number[]} nums
 * @returns {boolean}
 */
function containsDuplicate(nums: number[]): boolean {
  return nums.length !== [...new Set(nums)].length;
  // return nums.length !== new Set(nums).size; // We do not need to convert set to array in this case.
}

File: Easy/217 - ContainsDuplicate/Solution.cs
// Question 217. Contains Duplicate (https://leetcode-cn.com/problems/contains-duplicate/)

/// <summary>
/// 执行用时：136 ms, 在所有 C# 提交中击败了 56.51% 的用户
/// 内存消耗：31.8 MB, 在所有 C# 提交中击败了 20.56% 的用户
/// </summary>
public class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        Dictionary<int, bool> existence = new Dictionary<int, bool>();
        foreach (int num in nums)
        {
            if (existence.ContainsKey(num))
            {
                return true;
            }

            existence[num] = true;
        }

        return false;
    }
}